### PR TITLE
chore(ec2): add interface vpc endpoints

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

None

### Reason for this change

Some interface VPC endpoints are newly added.

### Description of changes

Add 3 interface VPC endpoints
- s3tables
- sagemaker-data-science-assistant'
- sagemaker-partner-app

### Description of how you validated changes

Executed AWS CLI:

```sh
$ aws ec2 describe-vpc-endpoint-services --filters Name=service-type,Values=Interface Name=owner,Values=amazon --region us-east-1 --query ServiceNames
    ...
    "aws.sagemaker.us-east-1.partner-app",
    ...
    "com.amazonaws.us-east-1.sagemaker-data-science-assistant",
    ...
    "com.amazonaws.us-east-1.s3tables",
```

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nmussy** [comment]
Hey @badmintoncryer, I ran the command and found the following endpoints also missing, if you want to add them as well:

- `com.amazonaws.savingsplans`
- `com.amazonaws.us-east-1.glue.dashboard`
- `com.amazonaws.us-east-1.invoicing`

**Comment by badmintoncryer** [comment]
@nmussy Thank you for your nice information😃 I'll add them later!

File: packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts [hunk 1]
```diff
@@ -562,12 +564,16 @@ export class InterfaceVpcEndpointAwsService implements IInterfaceVpcEndpointServ
   public static readonly S3 = new InterfaceVpcEndpointAwsService('s3');
   public static readonly S3_OUTPOSTS = new InterfaceVpcEndpointAwsService('s3-outposts');
   public static readonly S3_MULTI_REGION_ACCESS_POINTS = new InterfaceVpcEndpointAwsService('s3-global.accesspoint', 'com.amazonaws', undefined, { global: true });
+  public static readonly S3_TABLES = new InterfaceVpcEndpointAwsService('s3tables');
+  public static readonly SAVING_PLANS = new InterfaceVpcEndpointAwsService('savingsplans', 'com.amazonaws', undefined, { global: true });
```
Comment (nmussy) [inline packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts:568]: Missing plural:
```suggestion
  public static readonly SAVINGS_PLANS = new InterfaceVpcEndpointAwsService('savingsplans', 'com.amazonaws', undefined, { global: true });
```

Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts]: Oh! Thanks!!

**Review by nmussy — APPROVED** [review]
LGTM

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**